### PR TITLE
Rename a lot of stuff

### PR DESCRIPTION
**Pull request by tmeschter**
Rename a lot of stuff. Notably, I decided `IPropertyPageQueryCache` would be better named `IProjectState`, and most everything else is fall out from that.

There are no semantic changes.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/7202)

### CODE REVIEW

**Comment by tmeschter** [comment]
Apparently today is the day I decided I didn't like how I named things. :-)

**Review by dmonroym — APPROVED** [review]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/LaunchProfiles/SetLaunchProfilePropertyAction.cs [hunk 1]
```diff
@@ -28,7 +28,7 @@ public async Task ReceiveResultAsync(QueryProcessResult<IEntityValue> result)
             result.Request.QueryExecutionContext.CancellationToken.ThrowIfCancellationRequested();
             if (((IEntityValueFromProvider)result.Result).ProviderState is ContextAndRuleProviderState state)
             {
-                var cache = state.Cache;
+                var cache = state.ProjectState;
```
Comment (MiYanni) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/LaunchProfiles/SetLaunchProfilePropertyAction.cs:31]: `state` here is a `ProviderState` and `cache` should probably be `state`, I'd reckon. [or something of the sort]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/LaunchProfileDataProducer.cs [hunk 2]
```diff
@@ -11,7 +11,7 @@ namespace Microsoft.VisualStudio.ProjectSystem.VS.Query
 {
     internal static class LaunchProfileDataProducer
     {
-        public static IEntityValue CreateLaunchProfileValue(IQueryExecutionContext queryExecutionContext, EntityIdentity id, QueryProjectPropertiesContext propertiesContext, Rule rule, int order, IPropertyPageQueryCache cache, ILaunchProfilePropertiesAvailableStatus properties)
+        public static IEntityValue CreateLaunchProfileValue(IQueryExecutionContext queryExecutionContext, EntityIdentity id, QueryProjectPropertiesContext propertiesContext, Rule rule, int order, IProjectState cache, ILaunchProfilePropertiesAvailableStatus properties)
```
Comment (MiYanni) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Query/PropertyPages/LaunchProfileDataProducer.cs:14]: Probably need to find other instance variables named `cache` and make the name more appropriate if need be.

**Review by MiYanni — APPROVED** [review]
So, was it a cache before? Or, did it always just represent the state of something?

**Pull request merged**